forge-game/src/main/java/forge/game/GameActionUtil.java

```diff
@@ -566,7 +562,11 @@ public static SpellAbility addExtraKeywordCost(final SpellAbility sa) {
                 if (tr != null) {
                     String n = o.split(":")[1];
                     if (host.wasCast() && n.equals("X")) {
-                        n = Integer.toString(pc.announceRequirements(sa, "X for Casualty"));
+                        CardCollectionView creatures = CardLists.filter(CardLists.filterControlledBy(game.getCardsIn
+                                (ZoneType.Battlefield), activator), CardPredicates.Presets.CREATURES);
```

Comment:
Does this allow you to pick a number that you don't actually have a creature that you can sacrifice?

e.g. 1/1, 2/2, 4/4. But lets you choose 3?

Reply:
Right now it lets you pick anywhere in the range from your top power to your low power. Other Casualty cards let you sacrifice "creature with power _n_ or greater"... so I don't know if you could, say, pick X=3 and then sac a creature with power 4 in this case because you really want your copy to enter with 3 loyalty. The reminder text on this card seems a bit more strict, though... may need to revisit this again once we get more detailed rules.

Reply:
Gonna commit this to prevent NPE, and just add follow up update later to enhance this @Northmoc